akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -346,9 +349,12 @@ final class Split[T](decision: Split.SplitDecision, p: T ⇒ Boolean) extends Gr
             closeThis(handler, elem)
             handOver(handler)
           } else {
-            // Drain into the void
-            if (substreamCancelled) pull(in)
-            else substreamSource.push(elem)
+            if (substreamCancelled) {
+              if (eagerClose) completeStage()
```

Comment:
No, since we need to read at least one more message to find out whether substream has been fully consumed (if next message is matching predicate) or interrupted in the middle, since otherwise normal completion of first substream will always lead to master stream being cancelled on `eagerClose`.